src/workerd/api/basics.h

```diff
@@ -524,13 +527,15 @@ class AbortSignal final: public EventTarget {
       jsg::Optional<jsg::JsRef<jsg::JsValue>> maybeReason = kj::none,
       Flag flag = Flag::NONE);
 
+  class PendingReason: public kj::OneOf<kj::Array<kj::byte> /* v8Serialized*/,
+                           kj::Exception> /* if capability is dropped*/,
+                       public kj::Refcounted {};
```

Comment:
Just confirming that this is a safe way for the AbortTrigger rpc server and the AbortSignal object to share data.

Reply:
I don't think we should use inheritance this way.

There's actually a utility class that does exactly what you need here: `kj::RefcountedWrapper<OneOf<...>>`